packages/docsearch-react/src/DocSearch.tsx

```diff
@@ -42,24 +42,36 @@ export interface DocSearchProps {
   initialQuery?: string;
   navigator?: AutocompleteOptions<InternalDocSearchHit>['navigator'];
   translations?: DocSearchTranslations;
-  getMissingResultsUrl?: ({ query: string }) => string;
+  getMissingResultsUrl?: ({ query }: { query: string }) => string;
 }
 
 export function DocSearch(props: DocSearchProps) {
   const searchButtonRef = React.useRef<HTMLButtonElement>(null);
+  const activeElementRef = React.useRef<Element>(document.body);
```

Comment:
Is this working in server environments where `document` isn't defined?

Reply:
Interesting, I've never thought of rendering `DocSearch` in server side.

I guess it won't work.

Any suggestion?

Reply:
How about this:

<img width="2052" alt="" src="https://user-images.githubusercontent.com/1091472/210080133-0b243052-dfdb-4ee1-b09f-2014141fe4de.png">

I've reran the tests locally, seems fine.

Reply:
Isn't the effect covered by the `onOpen()` callback?

I think we better have `activeElementRef` accept `null` as initial value, and update its value in the `onOpen()` and `onClose()` callbacks. This way, the effect can also be removed, and there's no `document` in the initial render path.

```diff
 export function DocSearch(props: DocSearchProps) {
   const searchButtonRef = React.useRef<HTMLButtonElement>(null);
+  const activeElementRef = React.useRef<Element | null>(null);
   const [isOpen, setIsOpen] = React.useState(false);
   const [initialQuery, setInitialQuery] = React.useState<string | undefined>(
     props?.initialQuery || undefined
   );

   const onOpen = React.useCallback(() => {
     setIsOpen(true);
+    activeElementRef.current = document.activeElement || document.body;
   }, [setIsOpen]);

   const onClose = React.useCallback(() => {
     setIsOpen(false);
+
+    if (activeElementRef.current) {
+      activeElementRef.current.focus();
+      activeElementRef.current = null;
+    }
   }, [setIsOpen]);

-  React.useEffect(() => {
-    if (
-      !isOpen &&
-      activeElementRef.current &&
-      activeElementRef.current instanceof HTMLElement
-    ) {
-      activeElementRef.current.focus();
-    }
-  }, [isOpen]);

   // ...
 }
```

Reply:
Unfortunately this won't work,

```
   const onClose = React.useCallback(() => {
     setIsOpen(false);
+
+    if (activeElementRef.current) {
+      activeElementRef.current.focus();
+      activeElementRef.current = null;
+    }
   }, [setIsOpen]);
```

I believe I tried this in the beginning, and it would cause two tests to fail:

<img width="834" alt="" src="https://user-images.githubusercontent.com/1091472/210083068-a43de5bf-430f-46b6-a6ae-9c5b5281ed8f.png">

I.e., clicking the button to open dialog, then invoking shortcut to close the dialog would fail.